frontend/src/component/pure_module.rs

```diff
@@ -0,0 +1,42 @@
+use std::mem;
+use yew::{Component, ComponentLink, Html, ShouldRender, Properties};
+use mixlab_protocol::ModuleId;
+use crate::workspace::Window;
+
+pub trait PureModule: Clone + PartialEq + 'static {
+    fn view(&self, id: ModuleId, module: ComponentLink<Window>) -> Html;
+}
+
+#[derive(Properties, Clone)]
+pub struct PureProps<Params: PureModule> {
+    pub id: ModuleId,
```

Comment:
Did you want to drop ID but are waiting on refactoring its usage away from labels first?

Reply:
Yep